plugins/falco-agent/pkg/routes.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fairwindsops/controller-utils/pkg/controller"
+	"github.com/fairwindsops/insights-plugins/falco-agent/pkg/data"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+)
+
+const outputfolder = "/output"
+
+func inputDataHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	payload, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		logrus.Errorf("Error reading body: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	outputFile := fmt.Sprintf("%s/%s.json", outputfolder, strconv.FormatInt(time.Now().Unix(), 10))
+	err = ioutil.WriteFile(outputFile, []byte(payload), 0644)
+	if err != nil {
+		logrus.Errorf("Error writting to file: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Write([]byte(`{"status": "ok"}`))
+}
+
+func outputDataHandler(w http.ResponseWriter, r *http.Request, ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper) {
+	w.Header().Set("Content-Type", "application/json")
+	payload, err := data.Aggregate24hrsData(outputfolder)
+	if err != nil {
+		logrus.Errorf("Error while aggregating data: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	requestArray := make([]data.FalcoOutput, 0, len(payload))
+
+	workloads, err := controller.GetAllTopControllers(ctx, dynamicClient, restMapper, "")
```

Comment:
Per offline discussion, I think this will be more efficient if we just get one controller at a time for the pods we're interested in

Reply:
Actually - it would probably be better to gather this data when the events come _in_, and to save it to disk.

That way if a controller got deleted, we'll still have the data.